src/cluster/placement/algo/sharded_test.go

```diff
@@ -1306,6 +1306,42 @@ func TestBalanceShardsForShardedWhenImbalanced(t *testing.T) {
 	assert.Equal(t, expectedInstances, balancedPlacement.Instances())
 }
 
+func TestInitialPlacementIsBalanced(t *testing.T) {
+	instances := make([]placement.Instance, 0)
+
+	for i := 0; i < 10; i++ {
+		instances = append(instances,
+			placement.NewEmptyInstance(fmt.Sprintf("instance-0-%03d", i), "iso0", "zone", "endpoint", 1),
+			placement.NewEmptyInstance(fmt.Sprintf("instance-1-%03d", i), "iso1", "zone", "endpoint", 1),
+		)
+	}
+
+	ids := make([]uint32, 1024)
+	for i := range ids {
+		ids[i] = uint32(i)
+	}
+
+	algo := newShardedAlgorithm(placement.NewOptions())
+	p, err := algo.InitialPlacement(instances, ids, 2)
+	require.NoError(t, err)
+
+	var (
+		min = math.MaxInt32
+		max = math.MinInt32
+	)
+	for _, instance := range p.Instances() {
+		n := instance.Shards().NumShards()
+		if n < min {
+			min = n
+		}
+		if n > max {
+			max = n
+		}
+	}
+	require.True(t, max-min <= 1,
+		"The number of shards per instance differs by more than 1, min=%d max=%d", min, max)
```

Comment:
As this is a unit test for a single concrete case, I would simplify it by asserting that `n` is within certain range.
What is done here would be more appropriate for a property based test that @Antanukas has suggested (if you decide to go with that approach).

Reply:
https://github.com/m3db/m3/pull/4020/commits/1721594d34f54216415e58e57a196e1e00e149d2#diff-05cd0fd1dc4c1ee3969a247ba1ce3fb2fb21e609a49c08e470f0df5af02e88c5R1341-R1342